schema/deploy/roles/idle-session-disconnector/grants.sql

```diff
@@ -0,0 +1,33 @@
+-- Deploy seattleflu/id3c-customizations:roles/idle-session-disconnector/grants to pg
+
+begin;
+
+revoke all on database :"DBNAME" from "idle-session-disconnector";
+revoke all on schema receiving, warehouse, shipping from "idle-session-disconnector";
+revoke all on all tables in schema receiving, warehouse, shipping from "idle-session-disconnector";
+
+grant connect on database :"DBNAME" to "idle-session-disconnector";
+
+-- Regular users cannot see the state of sessions that they don't own. Create a
+-- new view using security definer with limited columns that this role can
+-- access to see a session's state.
+create or replace function get_stat_activity() returns table(
```

Comment:
For our setup, it's generally good to explicitly specify a schema name for functions/tables/view/etc when creating them. This makes it very clear where they will exist. Otherwise, they're created in the first schema listed in `search_path`, which might be variable across deploys.

The `public` schema is an ok place to put this function, but note that functions in the `public` schema are by default executable by anyone (i.e. the pseudo-role also named `public`). From the Pg 10 docs on GRANT:

> PostgreSQL grants default privileges on some types of objects to PUBLIC. No privileges are granted to PUBLIC by default on tables, table columns, sequences, foreign data wrappers, foreign servers, large objects, schemas, or tablespaces. For other types of objects, the default privileges granted to PUBLIC are as follows: CONNECT and TEMPORARY (create temporary tables) privileges for databases; EXECUTE privilege for functions; and USAGE privilege for languages and data types (including domains). The object owner can, of course, REVOKE both default and expressly granted privileges. (For maximum security, issue the REVOKE in the same transaction that creates the object; then there is no window in which another user can use the object.) Also, these initial default privilege settings can be changed using the ALTER DEFAULT PRIVILEGES command.

Since this is a sensitive function (security definer), we'll want to revoke `execute` privileges from everyone but `idle-session-disconnector`.

Reply:
I specified the public schema and revoked execute privileges from `public` before granting execute privileges to `idle-session-disconnector`.